.github/workflows/core-tests.yml

```
@@ -13,24 +13,19 @@ jobs:
       matrix:
         BUILD_TYPE: [ Release, Debug ]
     steps:
-    - uses: actions/checkout@v3
-    - name: Download libraries
-      uses: ./.github/composite-actions/download-libraries
-      with:
-        download-pybind: false
-    - name: Download datasets
-      uses: ./.github/composite-actions/download-datasets
-    - name: Configure CMake
-      run: >
-        cmake -B ${{github.workspace}}/build
-        -DCMAKE_C_COMPILER=gcc-10
-        -DCMAKE_CXX_COMPILER=g++-10
-        -DBOOST_ROOT=${{github.workspace}}/lib/boost
-        -DCMAKE_BUILD_TYPE=${{matrix.BUILD_TYPE}}
-        -Dgtest_disable_pthreads=OFF
-    - name: Build
-      run: cmake --build ${{github.workspace}}/build --config ${{matrix.BUILD_TYPE}}
-    - name: Test
-      working-directory: ${{github.workspace}}/build/target
-      shell: bash
-      run: ./Desbordante_test --gtest_filter='*:-*HeavyDatasets*'
+      - uses: actions/checkout@v3
+      - name: Download libraries
+        uses: ./.github/composite-actions/download-libraries
+      - name: Download datasets
+        uses: ./.github/composite-actions/download-datasets
+      - name: Build
+        run: |
+          if [[ "$BUILD_TYPE" == "Debug" ]]; then 
```

This isn't working, your Debug tests are compiled in Release mode.

Try this instead
```suggestion
          if [[ "${{matrix.BUILD_TYPE}}" == "Debug" ]]; then 
```